chia/wallet/trading/offer.py

```diff
@@ -152,7 +152,7 @@ def __post_init__(self) -> None:
             assert cs.coin not in adds
             try:
                 hinted_coins, cost = compute_spend_hints_and_additions(cs)
-                max_cost -= cost
+                max_cost = uint64(max_cost - cost)
```

Comment:
Similarly, what is the effect of raising here when less than 0 compared to existing behavior?